services/payments/secrets.go

```diff
@@ -260,17 +386,32 @@ func (s *Service) decryptSecrets(ctx context.Context) (map[string]string, error)
 		return nil, fmt.Errorf("failed to parse private key bytes for secret decryption: %w", err)
 	}
 
-	buf := bytes.NewBuffer(s.secretsCiphertext)
+	return age.Decrypt(bytes.NewReader(buf.Bytes()), identity)
+}
 
-	r, err := age.Decrypt(buf, identity)
+func (s *Service) encryptWithShares(ctx context.Context, data []byte) (io.Reader, error) {
+	// combine the service configured key shares
+	privateKey, err := shamir.Combine(s.keyShares)
 	if err != nil {
-		return nil, fmt.Errorf("failed to decrypt the secrets ciphertext: %w", err)
+		return nil, fmt.Errorf("failed to combine keyShares: %w", err)
 	}
 
-	var output = map[string]string{}
-	if err := json.NewDecoder(r).Decode(&output); err != nil {
-		return nil, fmt.Errorf("failed to json decode the secrets: %w", err)
+	identity, err := age.ParseX25519Identity(string(privateKey))
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse private key bytes for secret decryption: %w", err)
 	}
 
-	return output, nil
+	out := &bytes.Buffer{}
+
+	w, err := age.Encrypt(out, identity.Recipient())
+	if err != nil {
+		return nil, fmt.Errorf("Failed to create encrypted file: %v", err)
+	}
+	if _, err := io.WriteString(w, string(data)); err != nil {
+		return nil, fmt.Errorf("Failed to write to encrypted file: %v", err)
+	}
+	if err := w.Close(); err != nil {
+		return nil, fmt.Errorf("Failed to close encrypted file: %v", err)
+	}
+	return out, nil
```

Comment:
overall this looks good to me - I just added a few suggestions for better defense in depth from memory-based attacks